subprojects/core/src/main/java/org/gradle/internal/service/scopes/ExecutionGradleServices.java

```diff
@@ -170,7 +171,7 @@ public ExecutionEngine createExecutionEngine(
             new LoadPreviousExecutionStateStep<>(
             new MarkSnapshottingInputsStartedStep<>(
             new RemoveUntrackedExecutionStateStep<>(
-            new SkipEmptyWorkStep(outputChangeListener, skipEmptyWorkOutputsCleanerSupplier,
+            new SkipEmptyWorkStep(outputChangeListener, workInputListeners, skipEmptyWorkOutputsCleanerSupplier,
```

Comment:
`SkipEmptyWorkStep` does too much. We don't necessarily need to fix this in this PR, but we should at some point separate the broadcasting of inputs from it.